examples/example_tabular_regression.py

```diff
@@ -0,0 +1,116 @@
+"""
+======================
+Tabular Regression
+======================
+
+The following example shows how to fit a sample classification model
+with AutoPyTorch
+"""
+import os
+import tempfile as tmp
+import typing
+import warnings
+
+from sklearn.datasets import make_regression
+
+from autoPyTorch.data.tabular_feature_validator import TabularFeatureValidator
+
+os.environ['JOBLIB_TEMP_FOLDER'] = tmp.gettempdir()
+os.environ['OMP_NUM_THREADS'] = '1'
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
+
+warnings.simplefilter(action='ignore', category=UserWarning)
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+from sklearn import model_selection, preprocessing
+
+from autoPyTorch.api.tabular_regression import TabularRegressionTask
+from autoPyTorch.datasets.tabular_dataset import TabularDataset
+from autoPyTorch.utils.hyperparameter_search_space_update import HyperparameterSearchSpaceUpdates
+
+
+def get_search_space_updates():
+    """
+    Search space updates to the task can be added using HyperparameterSearchSpaceUpdates
+    Returns:
+        HyperparameterSearchSpaceUpdates
+    """
+    updates = HyperparameterSearchSpaceUpdates()
+    updates.append(node_name="data_loader",
+                   hyperparameter="batch_size",
+                   value_range=[16, 512],
+                   default_value=32)
+    updates.append(node_name="lr_scheduler",
+                   hyperparameter="CosineAnnealingLR:T_max",
+                   value_range=[50, 60],
+                   default_value=55)
+    updates.append(node_name='network_backbone',
+                   hyperparameter='ResNetBackbone:dropout',
+                   value_range=[0, 0.5],
+                   default_value=0.2)
+    return updates
+
+
+if __name__ == '__main__':
+    ############################################################################
+    # Data Loading
+    # ============
+
+    # Get the training data for tabular regression
+    # X, y = datasets.fetch_openml(name="cholesterol", return_X_y=True)
+
+    # Use dummy data for now since there are problems with categorical columns
+    X, y = make_regression(
+        n_samples=5000,
+        n_features=4,
+        n_informative=3,
+        n_targets=1,
+        shuffle=True,
+        random_state=0
+    )
+
+    X_train, X_test, y_train, y_test = model_selection.train_test_split(
+        X,
+        y,
+        random_state=1,
+    )
+
+    # Scale the regression targets to have zero mean and unit variance.
+    # This is important for Neural Networks since predicting large target values would require very large weights.
+    # One can later rescale the network predictions like this: y_pred = y_pred_scaled * y_train_std + y_train_mean
+    y_train_mean = y_train.mean()
```

Comment:
do you think it is better to do this in the target validator class?

Reply:
Open for discussion I would say. We could also put it in a separate component in the regression pipeline.